libkbfs/interfaces.go

```diff
@@ -1683,17 +1683,21 @@ type NodeCache interface {
 	UpdatePointer(oldRef BlockRef, newPtr BlockPointer) bool
 	// Move swaps the parent node for the corresponding Node, and
 	// updates the node's name.  NodeCache ignores the call when ptr
-	// is not cached.  Returns an error if newParent cannot be found.
-	// If newParent is nil, it treats the ptr's corresponding node as
-	// being unlinked from the old parent completely.
-	Move(ref BlockRef, newParent Node, newName string) error
+	// is not cached.  If newParent is nil, it treats the ptr's
+	// corresponding node as being unlinked from the old parent
+	// completely. If successful, it returns a function that can be
+	// called to undo the effect of the move (or `nil` if nothing
+	// needs to be done); if newParent cannot be found, it returns an
+	// error.
+	Move(ref BlockRef, newParent Node, newName string) (func(), error)
```

Comment:
Everywhere would be great! The reason is sometimes it's called using the actual type, and the completer would pull the concrete type's method signature